### ISSUE TITLE
Training Accuracy reported with on_batch_end method

### ISSUE THREAD

**Issue by oxsem** [keras-team/keras#9243]
Hi everyone, 

could anyone please help me to understand which accuracy is reported by **on_batch_end** method.
As you may know **on_batch_end** enables access accuracy as follows: _logs['acc']_.
At the same time with verbose=1 we can observe the loss and accuracy after every batch. When checking a particular example:

```
class AucCallback(Callback):
    def __init__(self, input_data=None, batch_size=256, collar="train"):
        super(Callback, self).__init__()
        self.batch_size = batch_size
        self.X, self.Y = input_data  

   def on_batch_end(self, batch, logs=None):
        print '\nBatch END Keras (accuracy): ', logs['acc']      

        # check accuracy
         loss_tr, acc_tr = self.model.evaluate(self.X, self.Y, verbose=0)
         print('Batch LOSS checkup. loss: {}, acc: {}\n'.format(loss_tr, acc_tr))

         # check AUC
         pred = self.model.predict(self.X, batch_size=self.batch_size, verbose=0)
         pred = np.transpose(pred)
         x = np.reshape(pred[1], (len(pred[1])))
         y = np.transpose(self.Y)[1]
         y = np.reshape(y.astype("int"), len(y))        
         print ("Batch AUC = %f\n" % (roc_auc_score(y, x)))
```

Here is the output of few batches:

> Epoch 1/5
> Batch END Keras (accuracy):  0.44921875
> BATCH AUC = 0.408232
> Batch LOSS checkup. loss: 0.816142838052, acc: 0.594060283688
> 512/7895 [>.............................] - ETA: 212s - loss: 0.6975 - acc: 0.4492
> Batch END Keras (accuracy):  0.443359375
> BATCH AUC = 0.419726
> Batch LOSS checkup. loss: 0.733189630001, acc: 0.594060283688
> 
> 1024/7895 [==>...........................] - ETA: 191s - loss: 0.6907 - acc: 0.4463
> Batch END Keras (accuracy):  0.490234375
> BATCH AUC = 0.448657
> Batch LOSS checkup. loss: 0.669601431637, acc: 0.594060283688
> 
> 1536/7895 [====>.........................] - ETA: 177s - loss: 0.6856 - acc: 0.4609
> Batch END Keras (accuracy):  0.51171875
> BATCH AUC = 0.683458
> Batch LOSS checkup. loss: 0.646700625893, acc: 0.66054964539
> 
> 2048/7895 [======>.......................] - ETA: 162s - loss: 0.6806 - acc: 0.4736
> Batch END Keras (accuracy):  0.45703125
> BATCH AUC = 0.771304
> Batch LOSS checkup. loss: 0.641690788032, acc: 0.481826241135
> ........

What could be the explanation for the fact that accuracy reported in progress bar or the one provided by _logs['acc']_ is not the same as when evaluating model manually in the end of every batch?

Thanks in advance!
